tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -0,0 +1,228 @@
+import pytest
+import os
+import json
+import logging
+
+from copy import deepcopy
+
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+DEVICES_PATH="usr/share/sonic/device"
+TIMEOUT=3600
+REBOOT_TYPES = {
+        "cold": "reboot",
+        "warm": "warm-reboot",
+        "fast": "fast-reboot"
+        }
+
+def find_pattern(lines, pattern):
+    for line in lines:
+        if pattern.match(line):
+            return True
+    return False
+
+def power_cycle(duthost=None, pdu_ctrl=None, delay_time=60):
+    if pdu_ctrl is None:
+        pytest.skip("No PSU controller for %s, skipping" % duthost.hostname)
+
+    all_outlets = pdu_ctrl.get_outlet_status()
+
+    for outlet in all_outlets:
+        pdu_ctrl.turn_off_outlet(outlet)
+    time.sleep(delay_time)
+    for outlet in all_outlets:
+        pdu_ctrl.turn_on_outlet(outlet)
+
+def reboot(duthost, pdu_ctrl, reboot_type):
+    if reboot_type == "power off": 
+        power_cycle(duthost, pdu_ctrl)
+        return
+
+    if reboot_type not in REBOOT_TYPES: pytest.fail("Invalid reboot type {}".format(reboot_type))
+
+    duthost.command(REBOOT_TYPES[reboot_type], module_ignore_errors=True, module_async=True)
+
+def complete_install(duthost, localhost, boot_type, res, pdu_ctrl, auto_reboot=False, current=None, next_image=None):
+    hn = duthost.mgmt_ip
+
+    if boot_type != "none":
+        if not auto_reboot:
+            logger.info("Waiting on install to finish.")
+            res.get(TIMEOUT)
```

Comment:
@alexrallen what if other vendors might need a longer/shorter timeout? Do we want to make it platform independent?

Reply:
Sure, support added to `firmware.json`